iree/base/api.c

```diff
@@ -0,0 +1,872 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iree/base/api.h"
+
+#include <assert.h>
+#include <limits.h>
+#include <stdarg.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+
+#include "iree/base/target_platform.h"
+#include "iree/base/tracing.h"
+
+static inline size_t iree_min_host_size(size_t a, size_t b) {
+  return a < b ? a : b;
+}
+
+#if IREE_PLATFORM_WINDOWS
+#define iree_aligned_alloc(alignment, size) _aligned_malloc(size, alignment)
+#define iree_aligned_free(p) _aligned_free(p)
+#else
+#define iree_aligned_alloc(alignment, size) aligned_alloc(alignment, size)
```

Comment:
Google integrate has errors here:

```
third_party/iree/iree/base/api.c:375:60: error: implicit declaration of function 'aligned_alloc' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
  iree_status_storage_t* storage = (iree_status_storage_t*)iree_aligned_alloc(
                                                           ^
third_party/iree/iree/base/api.c:36:45: note: expanded from macro 'iree_aligned_alloc'
#define iree_aligned_alloc(alignment, size) aligned_alloc(alignment, size)
                                            ^
third_party/iree/iree/base/api.c:433:60: error: implicit declaration of function 'aligned_alloc' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
  iree_status_storage_t* storage = (iree_status_storage_t*)iree_aligned_alloc(
                                                           ^
third_party/iree/iree/base/api.c:36:45: note: expanded from macro 'iree_aligned_alloc'
#define iree_aligned_alloc(alignment, size) aligned_alloc(alignment, size)
                                            ^
2 errors generated.
```

Reply:
Still happening even with `#include <stdlib.h>` on Android.

Can we add a fallback like https://github.com/Themaister/Granite/blob/ea8a9c1f549b6b924dd59cac69a860acb4e9f27d/util/aligned_alloc.cpp#L32-L58 ?